internal/pkg/addon/package.go

```diff
@@ -0,0 +1,276 @@
+	ObjectKeyProperty string
+
+	// ForceZip will force a zip file to be created even if the given file path
+	// points to a file. Directories are always zipped.
+	ForceZip bool
+}
+
+func (p *packagePropertyConfig) isStringReplacement() bool {
+	return len(p.BucketNameProperty) == 0 && len(p.ObjectKeyProperty) == 0
+}
+
+// resourcePackageConfig maps a CloudFormation resource type to configuration
+// for how to transform it's properties.
+//
+// This list of resources should stay in sync with
+// https://awscli.amazonaws.com/v2/documentation/api/latest/reference/cloudformation/package.html,
+// other than the AWS::Serverless resources, which are not supported in Copilot.
+//
+// TODO(dnrnd) AWS::Include.Location
+var resourcePackageConfig = map[string][]packagePropertyConfig{
+	"AWS::ApiGateway::RestApi": {
+		{
+			Property:           []string{"BodyS3Location"},
+			BucketNameProperty: "Bucket",
+			ObjectKeyProperty:  "Key",
+		},
+	},
+	"AWS::Lambda::Function": {
+		{
+			Property:           []string{"Code"},
+			BucketNameProperty: "S3Bucket",
+			ObjectKeyProperty:  "S3Key",
+			ForceZip:           true,
+		},
+	},
+	"AWS::Lambda::LayerVersion": {
+		{
+			Property:           []string{"Content"},
+			BucketNameProperty: "S3Bucket",
+			ObjectKeyProperty:  "S3Key",
+			ForceZip:           true,
+		},
+	},
+	"AWS::Serverless::Function": {
+		{
+			Property: []string{"CodeUri"},
+			ForceZip: true,
+		},
+	},
+	"AWS::Serverless::LayerVersion": {
+		{
+			Property: []string{"ContentUri"},
+			ForceZip: true,
+		},
+	},
+	"AWS::AppSync::GraphQLSchema": {
+		{
+			Property: []string{"DefinitionS3Location"},
+		},
+	},
+	"AWS::AppSync::Resolver": {
+		{
+			Property: []string{"RequestMappingTemplateS3Location"},
+		},
+		{
+			Property: []string{"ResponseMappingTemplateS3Location"},
+		},
+	},
+	"AWS::AppSync::FunctionConfiguration": {
+		{
+			Property: []string{"RequestMappingTemplateS3Location"},
+		},
+		{
+			Property: []string{"ResponseMappingTemplateS3Location"},
+		},
+	},
+	"AWS::Serverless::Api": {
+		{
+			Property: []string{"DefinitionUri"},
+		},
+	},
+	"AWS::ElasticBeanstalk::ApplicationVersion": {
+		{
+			Property:           []string{"SourceBundle"},
+			BucketNameProperty: "S3Bucket",
+			ObjectKeyProperty:  "S3Key",
+		},
+	},
+	"AWS::CloudFormation::Stack": {
+		{
+			// This implementation does not recursively package
+			// the local template pointed to by TemplateURL.
+			Property: []string{"TemplateURL"},
+		},
+	},
+	"AWS::Glue::Job": {
+		{
+			Property: []string{"Command", "ScriptLocation"},
+		},
+	},
+	"AWS::StepFunctions::StateMachine": {
+		{
+			Property:           []string{"DefinitionS3Location"},
+			BucketNameProperty: "Bucket",
+			ObjectKeyProperty:  "Key",
+		},
+	},
+	"AWS::Serverless::StateMachine": {
+		{
+			Property:           []string{"DefinitionUri"},
+			BucketNameProperty: "Bucket",
+			ObjectKeyProperty:  "Key",
+			ForceZip:           true,
```

Comment:
Oh why is this one `ForceZip`? compared to `AWS::StepFunctions::StateMachine`

Reply:
😅